raiden-ts/src/services/epics/udc.ts

```diff
@@ -238,27 +238,38 @@ export function udcAutoWithdrawEpic(
   {}: Observable<RaidenState>,
   { userDepositContract, address, config$, log }: RaidenEpicDeps,
 ): Observable<udcWithdraw.request | udcWithdrawPlan.success> {
-  let nextCheckBlock = 0;
+  let nextCheckAfter = 0;
+  const udcWithdrawTimeout$ = defer(async () =>
+    (await userDepositContract.callStatic.withdraw_timeout()).toNumber(),
+  ).pipe(retryWhile(intervalFromConfig(config$)), shareReplay({ bufferSize: 1, refCount: false }));
   return action$.pipe(
     filter(newBlock.is),
-    pluck('payload', 'blockNumber'),
-    filter((currentBlock) => currentBlock >= nextCheckBlock),
-    exhaustMap((currentBlock) =>
-      defer(async () => userDepositContract.withdraw_plans(address)).pipe(
-        mergeMap(function* ({ amount, withdraw_block: withdrawBlock }) {
+    filter(() => Date.now() >= nextCheckAfter),
+    exhaustMap(() =>
+      combineLatest([
+        defer(async () => userDepositContract.withdraw_plans(address)),
+        udcWithdrawTimeout$,
+      ]).pipe(
+        mergeMap(function* ([
+          { amount, withdrawable_after: withdrawableAfter },
+          udcWithdrawTimeout,
+        ]) {
           const meta = { amount: amount as UInt<32> };
-          if (withdrawBlock.isZero()) {
-            nextCheckBlock = currentBlock + UDC_WITHDRAW_TIMEOUT;
+          if (withdrawableAfter.isZero()) {
+            nextCheckAfter = Date.now() + udcWithdrawTimeout;
             return; // no plan, don't proceed to startupCheck
           }
-          const startupCheck = nextCheckBlock === 0;
+          const startupCheck = nextCheckAfter === 0;
           if (startupCheck) {
             yield of(
-              udcWithdrawPlan.success({ block: withdrawBlock.toNumber(), confirmed: true }, meta),
+              udcWithdrawPlan.success(
+                { withdrawableAfter: withdrawableAfter.toNumber() * 1e3, confirmed: true },
+                meta,
+              ),
             );
```

Comment:
This is the auto-withdraw epic for UDC. The idea here is that, if `config.autoUDCWithdraw=true`, we want to check both at startup and from time to time if there's a UDC plan ready to be withdrawn. Note also that we don't want concurrent checks, to avoid sending txs over and over while waiting, so we yield observables of actions and `mergeAll` them below. This `startupCheck` (first check) also must emit `udcWithdrawPlan.success`, which represents the success action emitted on the previous session or so.